### ISSUE TITLE
Future of Jill

### ISSUE THREAD

**Issue by abelsiqueira** [abelsiqueira/jill#69]
Hi all,

As you may know, [juliaup](https://github.com/JuliaLang/juliaup) ["will soon be announced as the official installer"](#33), so it is a good time to question the future of alternative installers - in particular, the one I created: Jill.
With that in mind, I would like to hear some feedback from users and contributors.

Here are some example questions, but feel free to reply with other opinions.

- Do you intend to keep using Jill?
- If yes, why? What are your reasons for using Jill? 
- Does `juliaup` solves the problem that led you to use `jill`?
- how do you download Jill? `curl`, `clone`, manually?
- how do you use Jill? Using `sudo`? Some config or none?
- how often do you update your Julia distribution - i.e., how ofter do you run Jill.
- how often do you update your Jill?
- Do you want more advanced features for `jill`? Such as #23 and #62?
- Would you contribute to Jill doing small issues? Such as updating the currently supported Julia version.
- Would you help maintain Jill?
- Would you fork Jill if it died?

Also, it appears Jill just passed 0xff stars. I really appreciate that you found this package useful - or at least interesting. Thank you very much :grinning: 

x-post:
-

**Comment by idevcde** [comment]
Hi! Just wanted to thank you very much for Jill. I have been using it extensively for a long time at the very beginning of my adventure with Julia. Currently I mostly use official binaries from Julia Language website. However, I still have Jill implemented in some of my scripts. My usage scenario is mostly related to installing Julia on various remote machines. Yes, I am still planning to use it. I have never used juliaup so I can’t compare. I like Jill as I am familiar with it, it almost always work and is very light. I mostly use curl. I rarely use it for updates. As I wrote, mostly for installs on clean machines. This is the prevailing scenario in my case. As for the frequencies it varies, sometimes a couple of times a day, sometimes once in a few weeks. As for advanced features, one scenario I found problematic was install on Arm machines, however, I have not explored this topic in depth. I would be happy to contribute in the scope of my current knowledge (pls be advised that I am not very experienced in coding). Thank you once again.

**Comment by omendezmorales** [comment]
Hi @abelsiqueira ,
I've used jill whenever I have switched laptops, and or needed to install the newest version of Julia (that interest me); I've used maybe once a quarter or so, and download it via _curl_. 
Until yesterday (07/27) had never heard of [juliaup](https://github.com/JuliaLang/juliaup), but seems also a nice utility. I hope you and David Anthoff might/want join forces, and keep doing the good work you've done so far.
Best wishes from 🇨🇱.

**Comment by mroavi** [comment]
Hi @abelsiqueira. I've been using Jill for a few years now, both on my local machine and on several servers I work with. I've never encountered any problems with it. However, since JuliaUp became the official installer, I decided to give it a try. There are two things I did not like, which made me revert to using Jill:

1. I don't appreciate installers modifying my .zshrc file. JuliaUp creates a machine-specific command to add its installation directory to the PATH environment variable. This is inconvenient because I like to version control my .zshrc file and the home directories on the different machines I use are different.

2. On top of that, JuliaUp has its internal bin folder. This is also not convenient since I prefer the more accepted convention of having locally installed binaries in the ~/.local/bin directory. This helps not having to add a new path to the PATH env var for every program I install.

Thanks for creating and maintaining it.

**Comment by nhz2** [comment]
I've recently created https://github.com/nhz2/install-julia-linux as another install script for julia. A key difference is that my script figures out the latest version using https://julialang-s3.julialang.org/bin/versions.json and verifies the download gpg like jill.py. This adds a lot of extra complexity to the script but means the script itself doesn't have to be updated often (only when the gpg key needs to be changed). So there is very little maintenance needed.

In terms of juliaup, from what I can tell it has a strong Windows first design philosophy so I do not recommend it for use on Linux currently.

**Comment by davidanthoff** [comment]
> In terms of juliaup, from what I can tell it has a strong Windows first design philosophy so I do not recommend it for use on Linux currently.

That is not correct, Juliaup is the official Julia installer on all platforms and supports Windows, Mac and Linux.

**Comment by nhz2** [comment]
Yes but obviously the defaults in one installer will never be optimal for everyone, and Julia is free so anyone can create and recommend an installer that has defaults that are optimal for their own use case and system.
